base/number.jl

```diff
@@ -11,6 +11,8 @@ length(x::Number) = 1
 endof(x::Number) = 1
 getindex(x::Number) = x
 getindex(x::Number, i::Integer) = i == 1 ? x : throw(BoundsError())
+getindex(x::Number, i1::Integer, i2::Integer) = 1 == i1 == i2 ? x : throw(BoundsError())
+getindex(x::Number, i1::Integer, i2::Integer, i3::Integer) = 1 == i1 == i2 == i3 ? x : throw(BoundsError())
```

Comment:
> "NUMBER OP ARRAY" and "ARRAY OP NUMBER"

Didn't SparseMatrixCSC already have specialized implementations for these? They should perhaps have been checking whether the number input was zero, but otherwise I'd expect them to be faster than the more general broadcast approach.